src/algebra/continued_fractions/computation/correctness_terminating.lean

```diff
@@ -0,0 +1,217 @@
+convergent `(gcf.of v).convergents n`. The residual term will be zero exactly when the continued
+fraction terminated; otherwise, the residual term will be given by the fractional part stored in
+`gcf.int_fract_pair.stream v n`.
+
+For an example, refer to `gcf.comp_exact_value` and for more information
+about the computation process, refer to `algebra.continued_fraction.computation.basic`.
+
+## Main definitions
+
+- `gcf.comp_exact_value` returns the exact value approximated by the continued fraction `gcf.of v`
+  by adding a residual term as described in the summary.
+
+## Main Theorems
+
+- `gcf.comp_exact_value_correctness_of_stream_eq_some` shows that `gcf.comp_exact_value` indeed
+  returns the value `v` when given the convergent and fractional part as described in the summary.
+- `gcf.of_correctness_of_terminated_at` shows the equality `v = (gcf.of v).convergents n`
+  if `gcf.of v` terminated at position `n`.
+-/
+
+namespace generalized_continued_fraction
+open generalized_continued_fraction as gcf
+
+variables {K : Type*} [discrete_linear_ordered_field K] {v : K} {n : ℕ}
+
+/--
+Given two successive continuants (at positions `n + 1` and `n`) of a continued fraction
+`generalized_continued_fraction.of v` and the fractional part at `int_fract_pair.stream n`,
+this function returns the exact value approximated by the continued fraction, that is `v`.
+
+For example, let `(v : ℚ) := 3.4`. We have
+- `gcf.int_fract_pair.stream v 0 = some ⟨3, 0.4⟩`, and
+- `gcf.int_fract_pair.stream v 1 = some ⟨2, 0.5⟩`.
+
+Now `(gcf.of v).convergents 1 = 3 + 1/2`, and our fractional term (the residual) at position `2` is
+`0.5`. We hence have `v = 3 + 1/(2 + 0.5) = 3 + 1/2.5 = 3.4`.
+-/
+protected def comp_exact_value (pred_conts conts : gcf.pair K) (fr : K) : K :=
+-- if the fractional part is zero, we exactly approximated the value by the last continuants
+if fr = 0 then conts.a / conts.b
+-- otherwise, we have to include the fractional part in a final continuants step.
+else let exact_conts := next_continuants 1 fr⁻¹ pred_conts conts in
+  exact_conts.a / exact_conts.b
+
+variable [floor_ring K]
+
+/-- Just a computational lemma we need for the next main proof. -/
+protected lemma comp_exact_value_correctness_of_stream_eq_some_aux_comp {a : K} (b c : K)
+  (fract_a_ne_zero : fract a ≠ 0) :
+  ((⌊a⌋ : K) * b + c) / (fract a) + b = (b * a + c) / fract a :=
+begin
+  have : ((⌊a⌋ : K) * b + c) / (fract a) + b = ((b * (fract a) + b * ⌊a⌋ + c) / fract a), by
+  { field_simp [fract_a_ne_zero], ac_refl },
+  simpa [←left_distrib] using this
+end
+
+open generalized_continued_fraction as gcf
+
+/--
+Shows the correctness of `comp_exact_value` in case the continued fraction did not terminate.
+That is, at every step of the computation of a continued fraction `gcf.of v`, we obtain the value
+`v` if we compute the continuant at position `n` and add the residual term given by the fractional
+part stored in `int_fract_pair.stream v n`.
```

Comment:
Same ambiguity as in the summary as to where you add the residual term. Also, should `continuant` be `convergent`?

Reply:
Fixed.